DNS: Allowing unmixed dual stack name servers

cathay4t opened the pull request:
This patch allowing configuring multiple dual stack DNS name servers
except placing IPv6/IPv4 name server in the middle of IPv4/IPv6 servers.

Test cases updated.

ffmancera commented on libnmstate/dns.py:
```diff
@@ -229,3 +230,21 @@ def _is_dns_config_changed(des_dns_state, cur_dns_state):
+    Return True when an IPv6 server is in the middle of two IPv4 namesevers or
+    an IPv4 server is in the middle of two IPv6 servers.
+    """
+    patten = ""
```
I think this variable name is a typo and it should be `pattern`.

ffmancera reviewed (changes requested):
Looks good to me, just a nit. Please, see my comment below. Thanks!

cathay4t committed:
DNS: Allowing unmixed dual stack name servers

This patch allowing configuring multiple dual stack DNS name servers
except placing IPv6/IPv4 name server in the middle of IPv4/IPv6 servers.

Test cases updated.

Signed-off-by: Gris Ge <fge@redhat.com>
